lib/spack/spack/version/version_types.py

```diff
@@ -917,45 +975,44 @@ def intersects(self, other):
 
         raise ValueError(f"Unsupported type {type(other)}")
 
-    def overlaps(self, other) -> bool:
-        return self.intersects(other)
-
-    def to_dict(self):
+    def to_dict(self) -> Dict:
         """Generate human-readable dict for YAML."""
         if self.concrete:
             return syaml_dict([("version", str(self[0]))])
         return syaml_dict([("versions", [str(v) for v in self])])
 
     @staticmethod
-    def from_dict(dictionary):
+    def from_dict(dictionary) -> "VersionList":
         """Parse dict from to_dict."""
         if "versions" in dictionary:
             return VersionList(dictionary["versions"])
         elif "version" in dictionary:
             return VersionList([Version(dictionary["version"])])
         raise ValueError("Dict must have 'version' or 'versions' in it.")
 
-    def update(self, other: "VersionList"):
-        for v in other.versions:
-            self.add(v)
+    def update(self, other: "VersionList") -> None:
+        self.add(other)
```

Comment:
This is more compact, but it is also an alias of `add` in the current implementation. 

I wonder, since the `update` operation takes just a `VersionList`, if the API was like that for performance reasons (in cases where you know it's a `VersionList` we avoid two checks on types).

Reply:
I think one additional initial type check on `VersionList` is in the noise, given that any call to it will call `add()`, which also type checks, multiple times.